tools/codegen/model.py

```diff
@@ -153,9 +153,17 @@ def from_yaml(ei: Dict[str, object], loc: 'Location') -> 'NativeFunction':
         """
         e = ei.copy()
 
+        cpp_no_default_args = e.pop('cpp_no_default_args', [])
+        assert isinstance(cpp_no_default_args, list)
+
         funcs = e.pop('func')
         assert isinstance(funcs, str), f'not a str: {funcs}'
-        func = FunctionSchema.parse(funcs)
+        func = FunctionSchema.parse(funcs, cpp_no_default_args)
+
+        defaulted_arguments = {a.name for a in func.schema_order_arguments()
+                               if a.default is not None}
+        invalid_args = set(cpp_no_default_args).difference(defaulted_arguments)
+        assert len(invalid_args) == 0, f'Invalid cpp_no_default_args: {invalid_args}'
```

Comment:
nit: Put this in `__post_init__`; it doesn't really matter too much the way the code exists today, but the invariant you're testing here would apply if someone manually constructed a NativeFunction, so it shouldn't live in the parsing code.

Reply:
The list of `cpp_no_default_args` only exists in the parse function. If you manually construct a NativeFunction, then you would say `Argument(..., cpp_no_default=True)` instead.